### PR TITLE
Ports: p7zip

### PR DESCRIPTION
**Pull request by cooljohnny3**
Port of p7zip. This port uses https://github.com/jinfeihan57/p7zip as it's source which is a fork from the sourceforge version. This version receives updates and has additional codecs.

### CODE REVIEW

File: Ports/p7zip/patches/fixes.patch [hunk 1]
```diff
@@ -0,0 +1,41 @@
+diff --git a/CPP/7zip/Archive/Zip/ZipItem.cpp b/CPP/7zip/Archive/Zip/ZipItem.cpp
```
Comment (gunnarbeutner) [inline Ports/p7zip/patches/fixes.patch:1]: This patch fails to apply for me:

```
[gunnar@nyx p7zip (p7z)]$ ./package.sh
Installing dependencies of p7zip!
Fetching p7zip!
Downloading URL: https://github.com/jinfeihan57/p7zip/archive/refs/tags/v17.04.tar.gz
p7zip-17.04.tar.gz already exists
Expecting sha256sum: ea029a2e21d2d6ad0a156f6679bd66836204aa78148a4c5e498fe682e77127ef
sha256sum(p7zip-17.04.tar.gz) = 'ea029a2e21d2d6ad0a156f6679bd66836204aa78148a4c5e498fe682e77127ef'
+ tar -xzf p7zip-17.04.tar.gz (nocd)
+ touch .p7zip-17.04.tar.gz_extracted
Patching p7zip!
+ patch -p1
patching file CPP/7zip/Archive/Zip/ZipItem.cpp
patching file CPP/7zip/CMAKE/7za/CMakeLists.txt
Hunk #1 FAILED at 357 (different line endings).
1 out of 1 hunk FAILED -- saving rejects to file CPP/7zip/CMAKE/7za/CMakeLists.txt.rej
patching file CPP/myWindows/config.h
[gunnar@nyx p7zip (p7z)]$
```

File: Ports/p7zip/patches/fixes.patch [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+index 353e895..3f2eda1 100644
+--- a/CPP/7zip/Archive/Zip/ZipItem.cpp
++++ b/CPP/7zip/Archive/Zip/ZipItem.cpp
+@@ -19,6 +19,7 @@
+ #include "../Common/ItemNameUtils.h"
+ 
+ #include "ZipItem.h"
++#include "strings.h"
```
Comment (gunnarbeutner) [inline Ports/p7zip/patches/fixes.patch:9]:
```suggestion
+#include <strings.h>
```

Including system headers should be done with `<>`.

**Review by gunnarbeutner — CHANGES_REQUESTED** [review]
Also, please squash those commits.

File: Ports/p7zip/patches/fixes.patch (hunk 1 above)
Comment (cooljohnny3) [reply Ports/p7zip/patches/fixes.patch:1]: The one that is failing is the CMakeLists.txt. It seems that the file that is downloaded has CRLF line endings but the patch has LF line endings. I can convert the file to have LF line endings with dos2unix but should I assume people have that installed?

**Comment by cooljohnny3** [comment]
Made the requested changes and I think the patch should be working now.

File: Ports/p7zip/package.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+install() {
+    cp -r "bin/Codecs" "${SERENITY_INSTALL_ROOT}/usr/local/bin"
+    cp "bin/7z_" "${SERENITY_INSTALL_ROOT}/usr/local/bin"
+    cp "bin/7z.so" "${SERENITY_INSTALL_ROOT}/usr/local/bin"
+    cp "bin/7za" "${SERENITY_INSTALL_ROOT}/usr/local/bin"
+    cp "bin/7zCon.sfx" "${SERENITY_INSTALL_ROOT}/usr/local/bin"
+    cp "bin/7zr" "${SERENITY_INSTALL_ROOT}/usr/local/bin"
+}
```
Comment (linusg) [inline Ports/p7zip/package.sh:32]: These seem to rely on `configure` `cd`'ing? Please don't do that as these steps can also be run in isolation. Instead, set the [`workdir`](https://github.com/SerenityOS/serenity/blob/master/Ports/README.md#workdir) variable and use [`run`](https://github.com/SerenityOS/serenity/blob/master/Ports/README.md#run) for the individual commands.

**Review by linusg — CHANGES_REQUESTED** [review]
Please remove the `.7z` file - I realise it's quite small, but since there's no software installed by default that can interact with it, it shouldn't be included in the first place.

Please also update the commit message and make sure to use proper spelling and punctuation. No need to mentioned that you squashed or fixed previous commits.

**Pull request merged**